external-crates/move/crates/move-package/src/resolution/dependency_graph.rs

```diff
@@ -1249,15 +1257,26 @@ impl DependencyGraph {
                 .and_then(|v| v.as_array_of_tables().cloned());
         }
 
-        use std::io::Seek;
-        let mut lock = LockFile::new(
-            install_dir,
-            self.manifest_digest.clone(),
-            self.deps_digest.clone(),
-        )?;
-        lock.flush()?;
-        lock.rewind()?;
-
+        let mut lock;
+        match lock_path {
+            // Get a handle to update an existing Move.lock.
+            // Since dependency graph updates are compatible across
+            // all Move.lock schema versions, we can rely on the existing version.
+            Some(lock_path) if lock_path.exists() => {
+                lock = LockFile::from(install_dir, &lock_path)?
+            }
+            // Initialize a lock file if no existing lock_path is set for this operation.
+            _ => {
+                use std::io::Seek;
+                lock = LockFile::new(
+                    install_dir,
+                    self.manifest_digest.clone(),
+                    self.deps_digest.clone(),
+                )?;
+                lock.flush()?;
+                lock.rewind()?;
+            }
+        }
```

Comment:
nit: slight preference to avoid the initialisation across multiple code paths in this code.

```suggestion
        let mut lock = match lock_path {
            // Get a handle to update an existing Move.lock.
            // Since dependency graph updates are compatible across
            // all Move.lock schema versions, we can rely on the existing version.
            Some(lock_path) if lock_path.exists() => {
                LockFile::from(install_dir, &lock_path)?
            }
            // Initialize a lock file if no existing lock_path is set for this operation.
            _ => {
                use std::io::Seek;
                let lock = LockFile::new(
                    install_dir,
                    self.manifest_digest.clone(),
                    self.deps_digest.clone(),
                )?;
                lock.flush()?;
                lock.rewind()?;
                lock
            }
        }
```